### ISSUE TITLE
ZenTimings closing itself - how to make debug report ?

### ISSUE THREAD

**Issue by mooms06** [irusanov/ZenTimings#43]
I'm trying to use it on an AMD 7840HS to see the timings but, apparently it's not supported, and the app says to run a debug report and sen it to you

![image](https://github.com/irusanov/ZenTimings/assets/25405517/2d496b9e-75d5-430a-aeb7-89810e923536)

But as soon as I click OK, it closes, so how can I run a debug report ?

HWiNFO is apparently able to read the timings, bit it is LPDDR5X 6400, so the frequency seems wrong.

![image](https://github.com/irusanov/ZenTimings/assets/25405517/011ca158-e8b4-4660-86a1-6cc8ea7b1e9e)

**Comment by irusanov** [comment]
Hi,
This APU model is not supported, indeed. I have no platform to test.
Have to change that message as it is not correct, you can't run a debug report without a supported processor. This needs to be added as e new functionality.

Attached is the current development version, there's a slight chance it works on Phoenix, but I have no way to test it myself.

[ZenTimings_v1.30.1107-debug.zip](https://github.com/irusanov/ZenTimings/files/12594785/ZenTimings_v1.30.1107-debug.zip)

**Comment by mooms06** [comment]
Hi,

Sorry for the late answer, I wasn't notified about your reply.

Yes, the v1.30.1107 is working:, with missing/false info:

![image](https://github.com/irusanov/ZenTimings/assets/25405517/298686fb-79ee-487e-be0c-0c20d0375af6)

Please find the debug report attached.

Thank you.

[Debug_Report_28283197.0074037.txt.zip](https://github.com/irusanov/ZenTimings/files/12864238/Debug_Report_28283197.0074037.txt.zip)

**Comment by irusanov** [comment]
@mooms06 There seem to be some problem with the noitifications or I have overlooked it.
Thanks for the debug report, this is unknown table version for me, will try to add it for the next 1.31 release.

**Comment by mooms06** [comment]
Version 1.31 is reporting more accurate information:


![image](https://github.com/irusanov/ZenTimings/assets/25405517/989c91a7-d3e5-467d-99c1-32550fa5e067)

**Comment by mooms06** [comment]
If I click on Tools => System Info, it crashes after two seconds.

**Comment by irusanov** [comment]
It's probably due to these missing values on the right. I haven't been able to read the greyed out values on my 6800HS. If I ever manage to fix it for it, then it would probably work for you as well. It's something about Lenovo as mine is Ideapad 15 Pro and I see you're using Lenovo, too.

**Comment by mooms06** [comment]
Again, sorry for the late answer, Github didn't send any notification about your reply.

Yes, debug is working, here is the report if needed:

[Debug_Report_28393674.9421841.txt.zip](https://github.com/irusanov/ZenTimings/files/13773492/Debug_Report_28393674.9421841.txt.zip)

**Comment by irusanov** [comment]
Thanks for the report.
Lenovo really messed up the AOD table. I have finally managed to retrieve it by an alternative method, but it's unusable.

You can try this build of ZenTimings, but the important values are missing in the table my Lenovo laptop bios returns.

[ZenTimings_v1.31.1212-debug.zip](https://github.com/irusanov/ZenTimings/files/13780999/ZenTimings_v1.31.1212-debug.zip)

It looks like this for me

![image](https://github.com/irusanov/ZenTimings/assets/7671877/d5698435-cf1a-495f-bc2b-b516d7fa1172)

**Comment by mooms06** [comment]
Looks like this for me:

![image](https://github.com/irusanov/ZenTimings/assets/25405517/62449291-e900-413d-a3c0-82b7abfce1a2)

[Debug_Report_28396414.0044209.txt.zip](https://github.com/irusanov/ZenTimings/files/13788367/Debug_Report_28396414.0044209.txt.zip)

**Comment by irusanov** [comment]
It's the stupid lenovo modifications. The same table version as yours work on an Asus laptop.
I guess you can try one more build, but I doubt there will be a difference.

[ZenTimings_v1.31.1213-debug.zip](https://github.com/irusanov/ZenTimings/files/13789210/ZenTimings_v1.31.1213-debug.zip)

If no change, then we can try the Registry.
Do you know how to work with the registry editor?

`Win + R -> regedit -> OK` then navigate to `Computer\HKEY_LOCAL_MACHINE\HARDWARE\ACPI\` and you should see bunch of folders. These are the ACPI tables and there should be several starting with `SSD`, expand one until there are no more levels. The IDs on my machine are `CB-01___`, but might be different on yours. One of these SSD entries will be the one we're looking for, but I need the identifier Lenovo decided to use to try and locate it. But it might be broken as mine and unusable.

![image](https://github.com/irusanov/ZenTimings/assets/7671877/85ff62cf-4568-46c0-ac8e-eefdddaf9d0c)

**Comment by mooms06** [comment]
Seems the same :
![image](https://github.com/irusanov/ZenTimings/assets/25405517/ece3fb24-2feb-4236-a20d-9950a898badc)



Which one you want ?

![image](https://github.com/irusanov/ZenTimings/assets/25405517/532a5f0d-28b1-4819-b555-eeb091e60e46)

Here is the first two:
[SSD1-2.zip](https://github.com/irusanov/ZenTimings/files/13789411/SSD1-2.zip)

**Comment by irusanov** [comment]
It's probably the SSD1 or SSD2 judging from the name.

**Comment by mooms06** [comment]
Then they are attached in the previous message.

**Comment by mooms06** [comment]
Unrelated: it appears LPDDR5 "6400" is actually 4x 1600 MT ? 
Since it's DDR, the actual frequency is 800 MHz, is this correct ? That's quite low.

Latest HWiNFO64 is now reporting 800 MHz:

![image](https://github.com/irusanov/ZenTimings/assets/25405517/501dcf08-e6bd-4aed-a494-c454d0501443)

On my desktop, it's reporting 1900 MHz for the DDR4 3800 so it seems accurate:

![image](https://github.com/irusanov/ZenTimings/assets/25405517/c1b7f428-6726-482a-95b8-aad85fe7b2c5)

**Comment by irusanov** [comment]
Can't find the signature on any of them, maybe Lenovo have changed that as well.

As for the speed, I'm not entirely sure, so I haven't changed the calculation yet. Currently it uses the DDR formula for the data rate.
for LPDDR5 it's more like 2 x IO Bus frequency as it is double data rate (2x3200), but the MCLK is reported as 800. I need to read some LPDDR specs to understand it better.

https://en.wikipedia.org/wiki/LPDDR

**Comment by irusanov** [comment]
I think we have to consider this formula

Data Transfer Rate (MT/s) = (Clock Frequency (MHz) / 2) x Data Bus Width Factor
the clock frequency is the MCLK and is reported by the bios, then we have to detect the mode which the current configuration uses and determine the factor

16-bit mode: 2
32-bit mode: 4
64-bit mode: 8

**Comment by mooms06** [comment]
According to Wikipedia about LPDDR5:

> The doubling of the transfer rate, and the **quarter-speed master clock**, results in a master clock which is half the frequency of a similar LPDDR4 clock. The command (CA) bus is widened to 7 bits, and commands are transferred at double data rate, so commands end up being sent at the same rate as LPDDR4.

Also, my Laptop is equipped with LPDDR5x according to Lenovo specs, but I don't know if there is a way to differentiate it from LPDDR5.

**Comment by mooms06** [comment]
HWiNFO64 is also reporting a 8:1 ratio:

![image](https://github.com/irusanov/ZenTimings/assets/25405517/e896ba84-ea44-438d-bdfa-3c6727859427)

**Comment by irusanov** [comment]
Yes, that's what it shows on mine as well (IdeaPad 5 Pro 16ARH7 with 6800HS and LPDDR5).
